File: frontend/src/app/Post/post.ts
import { Comment } from "../Comment/comment";

export class Post{
    id_post: Number;
    id_proffessional: Number;
    title: String;
    body: String;
    dt: String;
    comments: Comment[];

    constructor(id_proffessional: Number, title: String, body: String, dt: String){
        this.id_post = 0;
        this.id_proffessional = id_proffessional;
        this.title = title;
        this.body = body;
        this.dt = dt;
        this.comments = [];
    }
}

File: backend/src/main/java/com/project/linkedin/Post/Post.java
package com.project.linkedin.Post;


import lombok.Data;

import javax.persistence.*;


@Data
@Entity
public class Post {

    private @Id @GeneratedValue(strategy = GenerationType.SEQUENCE) Long id_post;
    private Long id_proffessional;
    private String title;
    private String body;
    private String dt;

    public Post() {
    }

    public Post(Long id_proffessional, String title, String  body, String dt) {

        this.id_proffessional = id_proffessional;
        this.title = title;
        this.body = body;
        this.dt = dt;

    }
}
